lib/thrift/protocol/binary.ex

```diff
@@ -121,6 +131,38 @@ defmodule Thrift.Protocol.Binary do
     {:error, {:cant_decode_message, rest}}
   end
 
+  def deserialize(:application_exception, binary) when is_binary(binary) do
```

Comment:
You'll need to update the `@spec deserialize` when you merge the upstream code. It's currently just looking for `:message_begin` as the first argument. We could just make that `atom` if there are going to be a variety of top-level deserialization types.